eva-server/src/main/java/uk/ac/ebi/eva/server/repository/VariantEntityRepositoryImpl.java

```diff
@@ -38,28 +40,104 @@
  * <p>It also implements the VariantEntityRepositoryCustom interface,
  * to provide an explicit implementation of the region query, using a margin for efficiency.
  */
+@Component
 public class VariantEntityRepositoryImpl implements VariantEntityRepositoryCustom {
 
-    private MongoDbFactory mongoDbFactory;
     private MongoTemplate mongoTemplate;
-    private MappingMongoConverter mappingMongoConverter;
 
     private final int MARGIN = 1000000;
 
     @Autowired
     public VariantEntityRepositoryImpl(MongoDbFactory mongoDbFactory, MappingMongoConverter mappingMongoConverter) {
-        this.mongoDbFactory = mongoDbFactory;
-        this.mappingMongoConverter = mappingMongoConverter;
         mongoTemplate = new MongoTemplate(mongoDbFactory, mappingMongoConverter);
     }
 
-    public List<VariantEntity> findByChrAndStartWithMarginAndEndWithMargin(String chr, int start, int end) {
+    void queryConsequenceType(Query query, List<String> consequenceType) {
+        List<Integer> consequenceTypeConv = consequenceType.stream()
+                                                           .map(c -> Integer.parseInt(c.replaceAll("[^\\d.]", ""), 10))
+                                                           .collect(Collectors.toList());
+        query.addCriteria(Criteria.where("annot.ct.so").in(consequenceTypeConv));
+    }
+
+    void relationalCriteriaHelper(Query query, String jsonPath, Double value,
+                                  VariantEntityRepository.RelationalOperator operator) {
+
+        Criteria criteria = Criteria.where(jsonPath);
+        switch (operator) {
+            case EQ:
+                criteria = criteria.is(value);
+                break;
+            case GT:
+                criteria = criteria.gt(value);
+                break;
+            case LT:
+                criteria = criteria.lt(value);
+                break;
+            case GTE:
+                criteria = criteria.gte(value);
+                break;
+            case LTE:
+                criteria = criteria.lte(value);
+                break;
+            case NONE:
```

Comment:
My suggestion for this method is:

1. Make it private.
2. Use `NONE` as condition to prevent a filter to be applied. This can be done either in:
   * `findByRegionAndComplexFilters`: replacing checks such as `if (mafValue != null && mafOperator != null)` with `if (mafOperator == VariantEntityRepository.RelationalOperator.NONE)`
   * `relationalCriteriaHelper`: Return the query object as it is. This would be less verbose but also less obvious to the reader.